utils/utils.go

```diff
@@ -301,13 +301,80 @@ func CurrencyBalance(
 
 	liveAmount, err := types.ExtractAmount(liveBalances, currency)
 	if err != nil {
-		return nil, nil, fmt.Errorf(
+		return nil, nil, nil, fmt.Errorf(
 			"%w: could not get %s currency balance for %s",
 			err,
 			types.PrettyPrintStruct(currency),
 			types.PrettyPrintStruct(account),
 		)
 	}
 
-	return liveAmount, liveBlock, nil
+	return liveAmount, liveBlock, liveCoins, nil
+}
+
+// AccountBalanceRequest defines the required information
+// to get an account's balance.
+type AccountBalanceRequest struct {
+	Account  *types.AccountIdentifier
+	Network  *types.NetworkIdentifier
+	Currency *types.Currency
+}
+
+// AccountBalance defines an account's balance,
+// including either balance or coins, as well as
+// the block which this balance was fetched at.
+type AccountBalance struct {
+	Account *types.AccountIdentifier
+	Amount  *types.Amount
+	Coins   []*types.Coin
+	Block   *types.BlockIdentifier
+}
+
+// GetAccountBalancesHelper is used by GetAccountBalances to determine
+// the CurrencyBalance at tip for an account
+type GetAccountBalancesHelper interface {
+	CurrencyBalance(
```

Comment:
nit: it feels weird to define an interface for another function in the same package. I think the better approach would be to replace the `fetcher` call in `CurrencyBalance` with a mock. (as a bonus we would also be testing that function)